### PR TITLE
[KARAF-5528] Stop refreshed bundles together with updated and deleted…

### PR DESCRIPTION
**Pull request by grgrzybek**
… ones

### CODE REVIEW

**Comment by grgrzybek** [comment]
@gnodet I checked with fabric8 itests. in this case I had problem with too early stopping of `dstate.serviceBundle`, thus this slight modification. Please verify

**Comment by grgrzybek** [comment]
@gnodet I checked with fabric8 itests. in this case I had problem with too early stopping of `dstate.serviceBundle`, thus this slight modification. Please verify

**Comment by grgrzybek** [comment]
retest this please

File: features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java [hunk 1]
```diff
@@ -633,6 +633,9 @@ public void deploy(DeploymentState dstate, DeploymentRequest request) throws Exc
             toStop.addAll(regionDeployment.toUpdate.keySet());
             toStop.addAll(regionDeployment.toDelete);
         }
+        Set<Bundle> toRefreshToStopEarly = new HashSet<>(toRefresh.keySet());
+        toRefreshToStopEarly.remove(dstate.serviceBundle);
+        toStop.addAll(toRefreshToStopEarly);
```
Comment (gnodet) [inline features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java]: Is there a need for this intermediary collection ?
What about 
```
toStop.addAll(toRefresh.keySet());
toStop.remove(serviceBundle);
```

**Comment by gnodet** [comment]
I thought the features service could support being stopped.
Do you recall what the problem was ?

File: features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java (hunk 1 above)
Comment (grgrzybek) [reply features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java]: intermediary was for case, when initial `toStop` contained service bundle. I didn't want to change that, as it worked.

**Comment by grgrzybek** [comment]
yes - stopping service bundle was supported. But in fabric8 case (with karaf 4 resolver), when toRefresh contained serviceBundle, I didn't want it to stop too early, as the process used it in between - it's probably not a problem in current Karaf4, but it *is* in fabric8v1

File: features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java (hunk 1 above)
Comment (gnodet) [reply features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java]: That's the part I don't get.
If it was supported, I don't see why you had a problem.  if it was not, we should not keep it.

**Pull request merged**